src/Tools/dotnet-trace/CommandLine/Commands/CollectCommand.cs

```diff
@@ -269,32 +280,59 @@ private static async Task<int> Collect(CancellationToken ct, IConsole console, i
                     using (VirtualTerminalMode vTermMode = printStatusOverTime ? VirtualTerminalMode.TryEnable() : null)
                     {
                         EventPipeSession session = null;
-                        try
+                        bool retry = true;
+                        while (retry)
                         {
-                            session = diagnosticsClient.StartEventPipeSession(providerCollection, collectRundownEvents, (int)buffersize);
-                            if (resumeRuntime)
+                            retry = false;
+                            try
                             {
-                                try
+                                EventPipeSessionConfiguration config = new(providerCollection, (int)buffersize, requestRundown: (rundownKeyword != 0), requestStackwalk: true, rundownKeyword: rundownKeyword);
+                                session = diagnosticsClient.StartEventPipeSession(config);
+                            }
+                            catch (UnsupportedCommandException e)
+                            {
+                                if (retryStrategy == RetryStrategy.DropKeywordKeepRundown)
                                 {
-                                    diagnosticsClient.ResumeRuntime();
+                                    Console.Error.WriteLine("The runtime is too old to support this tracing config, retrying with the standard rundown keyword");
+                                    Debug.Assert(rundownKeyword != 0);
+                                    Debug.Assert(rundownKeyword != EventPipeSession.DefaultRundownKeyword);
```

Comment:
Similar to comments on EventStreamProvider, all these asserts might trigger in supported scenarios.